airflow/providers/google/cloud/operators/dataflow.py

```diff
@@ -600,7 +600,7 @@ def __init__(
         options: dict[str, Any] | None = None,
         dataflow_default_options: dict[str, Any] | None = None,
         parameters: dict[str, str] | None = None,
-        location: str = DEFAULT_DATAFLOW_LOCATION,
+        location: str | None = None,
```

Comment:
Is this a breaking change or a bug fix?

Reply:
It's not breaking as the default location is assigned in hook when it is passes as null and there is no region defined (as far as I understand)

Reply:
I have the same concern, I think this needs to be tested

Reply:
It is bug fix. The additional tests are provided - the case is valid but will be broken without the fix. The default value is not necessary on the operator level, since it is being passed to the hook anyway (and it has default).

Reply:
Also additional level of protection can be provided at [_fallback_variable_parameter method](https://github.com/apache/airflow/blob/793d36f013efa191e7732faf6e624cdd1d83ebae/airflow/providers/google/cloud/hooks/dataflow.py#LL85C5-L85C33), but looks redundant since the default value is caught from the [DataflowHook.start_template_dataflow](https://github.com/apache/airflow/blob/793d36f013efa191e7732faf6e624cdd1d83ebae/airflow/providers/google/cloud/hooks/dataflow.py#L621) definition itself. Please let me know if it is useful - I can publish the PR

Reply:
My question was for clarification of this change in the release notes.
I have no objection either way